alchemist/alchemist-cognitive-agents/src/main/kotlin/it/unibo/alchemist/model/implementations/actions/Seek2D.kt

```diff
@@ -28,7 +29,7 @@ open class Seek2D<T, P>(
         return (current.surrounding(env, maxWalk) + superPosition)
             .asSequence()
             .discardUnsuitablePositions(env, pedestrian)
-            .minBy { it.distanceTo(super.target()) }?.minus(current)
-            ?: currentPosition
+            .minBy { it.distanceTo(target) }?.minus(current)
+            ?: env.origin()
```

Comment:
why?

Reply:
The next _relative_ position has to be returned: if we don't want to move, `env.origin()` (= the zero vector) is what we need. Returning `currentPosition` was an error.